src/App.tsx

```diff
@@ -1,24 +1,253 @@
+} from './api/todos';
+import { ErrorMessage } from './types/ErrorMessage';
+import { StatusFilter } from './types/StatusFilter';
+import { TodoInfo } from './components/TodoInfo';
+import { Header } from './components/header/Header';
+import { Footer } from './components/Footer/Footer';
+import { ErrorMessageItem }
+  from './components/ErrorMessageItem/ErrorMessageItem';
 
-const USER_ID = 0;
+const USER_ID = 10906;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState<
+  StatusFilter>(StatusFilter.ALL);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  const isToggleActive = todos.every(todo => todo.completed);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.LOAD);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => {
+        setErrorMessage(null);
+      }, 3000);
+    }
+  }, [errorMessage]);
+
+  const clearForm = () => {
+    setTempTodo(null);
+    setIsLoading(false);
+  };
+
+  const handleAddTodo = useCallback(async (newTitle: string) => {
+    if (!newTitle) {
+      setErrorMessage(ErrorMessage.TITLE);
+
+      return;
+    }
+
+    try {
+      setIsLoading(true);
+
+      const newTodoData = {
+        userId: USER_ID,
+        title: newTitle,
+        completed: false,
+      };
+
+      setTempTodo({ ...newTodoData, id: 0 });
+
+      const addedTodo = await addTodos(USER_ID, newTodoData);
+
+      setTodos((prevTodos => [...prevTodos, addedTodo]));
+      clearForm();
+    } catch {
+      clearForm();
+      setErrorMessage(ErrorMessage.ADD);
+    }
+  }, []);
+
+  const handleDeleteTodo = useCallback(async (todoId: number) => {
+    try {
+      setLoadingTodoIds((prevIds) => [...prevIds, todoId]);
+      await removeTodo(todoId);
+
+      setTodos((prevTodos => (
+        prevTodos.filter(todo => todo.id !== todoId)
+      )));
+
+      setLoadingTodoIds((prevIds) => (
+        prevIds.filter(prevId => prevId !== todoId)
+      ));
+    } catch {
+      setErrorMessage(ErrorMessage.REMOVE);
+    }
+  }, []);
+
+  const handleCloseError = () => {
+    setErrorMessage(null);
+  };
+
+  const activeTodos = useMemo(() => (
+    todos.filter(todo => !todo.completed)
+  ), [todos]);
+
+  const completedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed)
+  ), [todos]);
+
+  const visibleTodos = useMemo(() => {
+    const { ACTIVE, COMPLETED } = StatusFilter;
+
+    switch (selectedFilter) {
+      case ACTIVE:
+        return activeTodos;
+
+      case COMPLETED:
+        return completedTodos;
+
+      default:
+        return todos;
+    }
+  }, [todos, selectedFilter]);
```

Comment:
Let's use the filter in here and move this to a separate helper file
Then for activeTodos and completedTodos variables, use your helper function